src/app/collective-rewards/rewards/hooks/useGetLastCycleDistribution.test.tsx

```diff
@@ -0,0 +1,142 @@
+import {
+  useGetRewardDistributionFinishedLogs,
+  useGetLastCycleDistribution,
+  RewardDistributionFinishedEventLog,
+} from '@/app/collective-rewards/rewards'
+import { describe, expect, it, vi } from 'vitest'
+import { Cycle } from '@/app/collective-rewards/metrics'
+import { DateTime, Duration } from 'luxon'
+
+vi.mock('@/app/collective-rewards/rewards/hooks/useGetRewardDistributionFinishedLogs', () => {
+  return {
+    useGetRewardDistributionFinishedLogs: vi.fn(),
+  }
+})
+
+describe('useGetLastCycleRewardsTimestamps', () => {
+  type Log = RewardDistributionFinishedEventLog[number] & { timeStamp: number }
+  const startTimestamp = 1733011200 // 2024-12-01 00:00:00
+  const duration = 1209600 //  14 days
+  const distributionWindow = 3600 // 1 hour
+  const cycle: Cycle = {
+    cycleStart: DateTime.fromSeconds(startTimestamp),
+    cycleNext: DateTime.fromSeconds(startTimestamp + duration),
+    cycleDuration: Duration.fromObject({ seconds: duration }),
+    fistCycleStart: DateTime.fromSeconds(startTimestamp),
+    endDistributionWindow: DateTime.fromSeconds(distributionWindow),
+  }
+  const { cycleStart, cycleNext, cycleDuration } = cycle
+
+  it('should return (from,to) = to next cycle start if there are no events', () => {
```

Comment:
Why from should be equal to to? I think it's worth clarifying the reason either here or in the code with a comment